[ArcListView] Add highlight to item on click

grzegorz-czajkowski opened the pull request:
Issue: N/A

Problem: if user clicked not active element on list, then back button is pressed,
	highlight on item is not shown *sometimes*.

Cause: click event triggers 'Change' event but it may not be handled since 'ui-page-active'
	goes to new page element (sub page) and events from old page (not active) are ignored.

Solution: select item on 'pageshow' event

Remarks: on device, highlight sometimes becomes higher for a while but this is separate issue
	which is going to be handled separately.

Signed-off-by: Grzegorz Czajkowski <g.czajkowski@samsung.com>

grzegorz-czajkowski committed:
[ArcListView] Add highlight on "Back" button

Issue: N/A

Problem: if user clicked not active element on list, then back button is pressed,
	highlight on item is not shown *sometimes*.

Cause: click event triggers 'Change' event but it may not be handled since 'ui-page-active'
	goes to new page element (sub page) and events from old page (not active) are ignored.

Solution: select item on 'pageshow' event

Remarks: on device, highlight sometimes becomes higher for a while but this is separate issue
	which is going to be handled separately.

Signed-off-by: Grzegorz Czajkowski <g.czajkowski@samsung.com>

TomaszLukawskiSam reviewed (commented):
I verified this PR on device. Works good. There is issue with size of highlight but this seems be also before this PR but before was no possibility to see it.

singa2000 commented:
I verified the PR  in the Emulator, it looks good working.

grzegorz-czajkowski committed:
[ArcListView] Remove highlight on pagehide

Issue: N/A

Problem: on device, if user clicked not active element on list, then back button is pressed
	highlight sometimes becomes higher for a while

Cause: if all events are processed successfully on old page during switch to new page
	highlight is set with wrong dimensions (before animation start).

Solution: remove highlight since it will be restored on 'pageshow' event.

Signed-off-by: Grzegorz Czajkowski <g.czajkowski@samsung.com>

grzegorz-czajkowski commented:
@TomaszLukawskiSam added separate commit to this PR which fixes wrong highlight dimension on device.

TomaszLukawskiSam reviewed (approved):
This patch works very good. Thanks.
